### PR TITLE
Add algebraic types support for query nodes

### PR DESCRIPTION
**Pull request by dzhelezov**
This PR implements as described in https://github.com/Joystream/joystream/issues/709

In a nutshell, the following constructs are supported in the input schema:

```gql
type Miserable @variant {
  hates: String!
}

type HappyPoor @variant {
  isMale: boolean
}

union Poor = HappyPoor | Miserable

type MiddleClass @variant {
  father: Poor
  mother: Poor
}

type Rich @variant {
  bank: EntityC
}

union Status = Poor | MiddleClass | HappyPoor | Miserable

type Account @entity {
  id: ID!
  status: Status!
}
```

Such a construct would generate a query of the type 
```gql
query {
	accounts(limit: 5, orderBy: about_ASC, where: { status_json: { father: { isMale_eq: true }} }) {
    about
    status {
      __typename 
      ... on MiddleClass {
        father {
          ... on HappyPoor {
            isMale
          }
          ... on Miserable {
            hates
          }
        }
      }
    }
  }
}
```

Project Issue: https://github.com/Joystream/joystream/issues/819

### CODE REVIEW

File: query-node/substrate-query-framework/cli/src/templates/entities/model.ts.mst [hunk 1]
```diff
@@ -91,5 +102,16 @@ export {{#isInterface}}abstract{{/isInterface}} class {{className}}
       {{#description}}description: `{{{description}}}`{{/description}} })
     {{camelName}}{{^required}}?{{/required}}{{#required}}!{{/required}}:{{tsType}} 
   {{/is.enum}}
+
+
+  {{#is.union}} 
+	@Column('jsonb')
```
Comment (metmirr) [inline query-node/substrate-query-framework/cli/src/templates/entities/model.ts.mst:126]: Could you use `@CustomField({ db: { type: 'jsonb' }, api: { type: 'json' } })` ?

Comment (dzhelezov) [reply query-node/substrate-query-framework/cli/src/templates/entities/model.ts.mst:126]: Based on what I have researched `api: { type: 'json' }` would make the resulting (scalar) GraphQL type `JSON`, while here we preserve more structure by making it a union type

Comment (dzhelezov) [reply query-node/substrate-query-framework/cli/src/templates/entities/model.ts.mst:126]: So the query _input_ field is JSON, while the result is of Union type and strongly typed

Comment (metmirr) [reply query-node/substrate-query-framework/cli/src/templates/entities/model.ts.mst:126]: Okay! Got it! 👍

**Review by metmirr — APPROVED** [review]

**Pull request merged**